core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -394,6 +394,40 @@ public void redeployGatewayCertificate(@Nonnull Stack stack, @Nonnull Cluster cl
         }
     }
 
+    public void redeployGatewayPillarOnly(@Nonnull Stack stack, @Nonnull Cluster cluster) {
+        try {
+            Set<Node> allNodes = stackUtil.collectNodes(stack);
+            Set<Node> reachableNodes = stackUtil.collectReachableNodes(stack);
+            List<GatewayConfig> gatewayConfigs = gatewayConfigService.getAllGatewayConfigs(stack);
+            List<GrainProperties> grainsProperties = grainPropertiesService.createGrainProperties(gatewayConfigs, cluster, reachableNodes);
+            SaltConfig saltConfig = createSaltConfigWithGatewayPillarOnly(stack, cluster, grainsProperties);
+            ExitCriteriaModel exitCriteriaModel = clusterDeletionBasedModel(stack.getId(), cluster.getId());
+            LOGGER.debug("Calling orchestrator to upload gateway pillar");
+            hostOrchestrator.uploadGatewayPillar(gatewayConfigs, allNodes, exitCriteriaModel, saltConfig);
+        } catch (CloudbreakOrchestratorCancelledException e) {
+            LOGGER.debug("Orchestration cancelled during redeploying gateway pillar", e);
+            throw new CancellationException(e.getMessage());
+        } catch (CloudbreakOrchestratorException | IOException e) {
+            LOGGER.debug("Orchestration exception during redeploying gateway pillar", e);
+            throw new CloudbreakServiceException(e.getMessage(), e);
+        }
+    }
+
+    public void redeployStates(@Nonnull Stack stack, @Nonnull Cluster cluster) {
```

Comment:
I wouldn't use @Nonnull since it throws NPE that is not as sophisticated as throwing an IllegalArgumentException through our NullUtil's throwIfNull, like

`throwIfNull(stack, () -> new IllegalStateException("Stack should not be null!"));`
`throwIfNull(cluster, () -> new IllegalStateException("Cluster should not be null!"));`

Yeah it's a bit more line and seems like a "duplication" but based on my experiences, NPE is usually caused by some unexpected/-handled issue.

Not to mention that the name of IllegalArgumentException is much more descriptive

Reply:
ok